volumes/csi-wrapper/hack/azure/README.md

```diff
@@ -0,0 +1,96 @@
+# csi-wrapper
+Azure File CSI Wrapper for Peer Pod Storage
+
+## High Level Design
+
+![design](./images/csi-wrapper.png)
+
+> **Note** Edited via https://excalidraw.com/
+
+## Test CSI Wrapper with azurefiles-csi-driver for peer pod demo on Azure Cloud
+
+### Set up a demo environment on your development machine
+
+Follow the [README.md](../../azure/README.md) to setup a x86 based demo environment on Azure Cloud.
+
+### Deploy azurefiles-csi-driver on the worker node
+Note: All the steps can be performed anywhere with cluster access
+
+1. Login to the worker node
+```bash
+ssh root@ip-of-your-worker-node
+```
+
+2. Clone the source code on the worker node:
+```bash
+git clone https://github.com/kubernetes-sigs/azurefile-csi-driver
+cd azurefile-csi-driver
+```
+3. Get your worker instance id
+- Check your worker node name
+```bash
+kubectl get nodes -o wide
+```
+4. Run the script:
+```bash
+bash ./deploy/install-driver.sh master local
+```
+
+### Deploy csi-wrapper to patch on azurefiles-csi-driver
+
+1. Create the PeerpodVolume CRD object
+```bash
+cd /root/cloud-api-adaptor/
+kubectl create -f volumes/csi-wrapper/crd/peerpodvolume.yaml
+```
+The output looks like:
+```bash
+customresourcedefinition.apiextensions.k8s.io/peerpodvolumes.peerpod.azure.com created
+```
+2. Create vpc-block-csi-wrapper-runner role  bind to azurefiles-controller-sa account
+```bash
+kubectl create -f volumes/csi-wrapper/hack/hack/azure-files-csi-wrapper-runner.yaml
+```
+3. Build csi-wrapper images on worker node:
+```bash
+apt install docker.io -y
+cd volumes/csi-wrapper/
+make import-csi-controller-wrapper-docker
```

Comment:
if we have public `csi-controller-wrapper`, `csi-node-wrapper` and `csi-podvm-wrapper` docker images, this step can be skipped.
I have create a PR to build and push those csi-wrapper docker images, https://github.com/confidential-containers/cloud-api-adaptor/pull/1031